src/pages/send-bulk-message.js

```diff
@@ -0,0 +1,216 @@
+    const router = useRouter();
+		const user = unsafeExtractUser()
+    const [assignedCallbacks ,setAssignedCallbacks]  = useState(null)
+    const [unAssignedCallbacks ,setUnAssignedCallbacks]  = useState(null)
+    const [previewTemplate, setTemplatePreview] = useState([])
+    const [unassigned, setUnassigned] = useState({
+      value:false,
+      sendSms:false
+    })
+    const [assigned, setAssigned] = useState({
+      value:false,
+      sendSms:false
+    })
+    const [helpType, setHelpTpye] = useState(null)
+    const [callbacks, setCallbacks] = useState(null)
+    const [dropdownItems, setDropDown] =  useState([""])
+    const [errorsExist, setErrorsExist] = useState(false)
+    const getCallbacks = async () => {
+      const callbackGateway = new CallbackGateway();
+      let callbacks = await callbackGateway.getCallback({});
+      setCallbacks(callbacks)
+    }
+
+    const getPreview = async () => {
+      let govNotifyGateway = new GovNotifyGateway
+      let response = await govNotifyGateway.getTemplatePreview(PRE_CALL_MESSAGE_TEMPLATE)
+      setTemplatePreview(response.body)
+    }
+
+    useEffect(()=> {
+      getPreview()
+    }, [])
+    useEffect(()=> {
+      getCallbacks()
+    }, [])
+    useEffect(()=>{
+      setDropDown(["Please choose"].concat(callTypes))
+    }, [])
+    
+
+    const handleReview = async (event) => {
+      if((assigned.value || unassigned.value) && helpType){
+        event.preventDefault();
+        let govNotifyGateway = new GovNotifyGateway
+        await govNotifyGateway.sendBulkText(JSON.stringify({assigned:assigned, unassigned: unassigned, user:user.name, helpType: helpType, textTemplateId: PRE_CALL_MESSAGE_TEMPLATE}))
+        router.push('/dashboard');
+      }else{
+        setErrorsExist(true)
+      }
+    };
+
+    const updateSendTextMessageCases = (value) =>{
+      if (value == 'unassigned'){
+        let unassignedObject = (unassigned.value) ? {value: false, sendSms: false } : { value: true, sendSms: true}
+        setUnassigned(unassignedObject)
+        
+      } else if (value == 'assigned'){
+        let assignedObject = (assigned.value) ? {value: false, sendSms: false } : {value: true,sendSms: true }
+        setAssigned(assignedObject)
+      }
+    }
+
+    const updateHelpType = (value) => {
+      setHelpTpye(value)
+      let unassignedCallbacks = [];
+      let assignedCallbacks = [];
+      callbacks.forEach((callback) => {
+        if (callback.assignedTo == null && callback.callType == value || !callback.assignedTo && callback.callType == value ) {
+          unassignedCallbacks.push(callback);
+      } else if(callback.callType == value ) {
+          assignedCallbacks.push(callback);
+      } else if(value == 'All'){
+        if (callback.assignedTo == null || !callback.assignedTo) {
+          unassignedCallbacks.push(callback);
+        } else {
+          assignedCallbacks.push(callback);
+        }
+      }
+      });
```

Comment:
You can write this using map, but that's just personal preference. No need to change
```
if (value=="All") {
const unassignedCallbacks = callbacks.map(x => x.assignedTo==null || !callback.assignedTo)
const assignedCallbacks = callbacks.map(x => x.assignedTo!=null)
}
else{
const unassignedCallbacks = callbacks.map(x => (x.assignedTo==null || !callback.assignedTo) && callback.callType==value)
const assignedCallbacks = callbacks.map(x => x.assignedTo!=null && callback.callType==value)
}
```

Reply:
This might also not be 100% correct in terms of logic

Reply:
it think this just evaluates to true or false and returns that for each